Resource Function attribute timeout failing

<!---
Please note the following potential times when an issue might be in Terraform core:

* [Configuration Language](https://www.terraform.io/docs/configuration/index.html) or resource ordering issues
* [State](https://www.terraform.io/docs/state/index.html) and [State Backend](https://www.terraform.io/docs/backends/index.html) issues
* [Provisioner](https://www.terraform.io/docs/provisioners/index.html) issues
* [Registry](https://registry.terraform.io/) issues
* Spans resources across multiple providers

If you are running into one of these scenarios, we recommend opening an issue in the [Terraform core repository](https://github.com/hashicorp/terraform/) instead.
--->

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Terraform Version

<!--- Please run `terraform -v` to show the Terraform core version and provider version(s). If you are not running the latest version of Terraform or the provider, please upgrade because your issue may have already been fixed. [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions). --->

### Affected Resource(s)

<!--- Please list the affected resources and data sources. --->

* scaleway_function

### Terraform Configuration Files

<!--- Information about code formatting: https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code --->

```hcl
resource scaleway_function main {
  namespace_id = scaleway_function_namespace.main.id
  runtime      = "go118"
  handler      = "Handle"
  privacy      = "private"
  zip_file = "function.zip"
  timeout = "100s"
  zip_hash = filesha256("function.zip")
  deploy = true
}
```

### Debug Output


```
panic: interface conversion: interface {} is string, not int64
```

### Panic Output

<!--- If Terraform produced a panic, please provide a link to a GitHub Gist containing the output of the `crash.log`. --->

### Expected Behavior

<!--- What should have happened? --->

### Actual Behavior

<!--- What actually happened? --->

### Steps to Reproduce

<!--- Please list the steps required to reproduce the issue. --->

1. `terraform apply`

### Important Factoids

<!--- Are there anything atypical about your accounts that we should know? --->

### References

<!---
Information about referencing Github Issues: https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests

Are there any other GitHub issues (open or closed) or pull requests that should be linked here? Vendor documentation? For example:
--->

* #0000

--- a/docs/resources/function.md
+++ b/docs/resources/function.md
@@ -42,12 +42,12 @@ resource scaleway_function main {
   runtime      = "go118"
   handler      = "Handle"
   privacy      = "private"
-  zip_file = "function.zip"
-  zip_hash = filesha256("function.zip")
-  deploy = true
+  timeout      = 10
+  zip_file     = "function.zip"
+  zip_hash     = filesha256("function.zip")
+  deploy       = true
 }
 ```
-
 
 ## Arguments Reference
 

--- a/scaleway/resource_function.go
+++ b/scaleway/resource_function.go
@@ -98,7 +98,7 @@ func resourceScalewayFunction() *schema.Resource {
 				Description: "Handler of the function. Depends on the runtime https://developers.scaleway.com/en/products/functions/api/#create-a-function",
 			},
 			"timeout": {
-				Type:        schema.TypeString,
+				Type:        schema.TypeInt,
 				Computed:    true,
 				Description: "Holds the max duration (in seconds) the function is allowed for responding to a request",
 				Optional:    true,
@@ -266,7 +266,7 @@ func resourceScalewayFunctionRead(ctx context.Context, d *schema.ResourceData, m
 	_ = d.Set("name", f.Name)
 	_ = d.Set("privacy", f.Privacy.String())
 	_ = d.Set("region", f.Region.String())
-	_ = d.Set("timeout", flattenDuration(f.Timeout.ToTimeDuration()))
+	_ = d.Set("timeout", f.Timeout.Seconds)
 	_ = d.Set("domain_name", f.DomainName)
 
 	return diags

--- a/scaleway/resource_function_test.go
+++ b/scaleway/resource_function_test.go
@@ -78,6 +78,41 @@ func TestAccScalewayFunction_Basic(t *testing.T) {
 	})
 }
 
+func TestAccScalewayFunction_Timeout(t *testing.T) {
+	tt := NewTestTools(t)
+	defer tt.Cleanup()
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: tt.ProviderFactories,
+		CheckDestroy:      testAccCheckScalewayFunctionDestroy(tt),
+		Steps: []resource.TestStep{
+			{
+				Config: `
+					resource scaleway_function_namespace main {}
+
+					resource scaleway_function main {
+						name = "foobar"
+						namespace_id = scaleway_function_namespace.main.id
+						runtime = "node14"
+						privacy = "private"
+						handler = "handler.handle"
+						timeout = 10
+					}
+				`,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckScalewayFunctionExists(tt, "scaleway_function.main"),
+					resource.TestCheckResourceAttr("scaleway_function.main", "name", "foobar"),
+					resource.TestCheckResourceAttr("scaleway_function.main", "runtime", "node14"),
+					resource.TestCheckResourceAttr("scaleway_function.main", "privacy", "private"),
+					resource.TestCheckResourceAttr("scaleway_function.main", "handler", "handler.handle"),
+					resource.TestCheckResourceAttr("scaleway_function.main", "timeout", "10"),
+				),
+			},
+		},
+	})
+}
+
 func TestAccScalewayFunction_NoName(t *testing.T) {
 	tt := NewTestTools(t)
 	defer tt.Cleanup()

--- a/scaleway/testdata/function-timeout.cassette.yaml
+++ b/scaleway/testdata/function-timeout.cassette.yaml
@@ -1,0 +1,535 @@
+---
+version: 1
+interactions:
+- request:
+    body: '{"name":"tf-func--sad-wilson","environment_variables":{},"project_id":"ee7bd9e1-9cbd-4724-b2f4-19e50f3cf38b","description":null,"secret_environment_variables":null}'
+    form: {}
+    headers:
+      Content-Type:
+      - application/json
+      User-Agent:
+      - scaleway-sdk-go/v1.0.0-beta.7+dev (go1.19.1; linux; amd64) terraform-provider/develop
+        terraform/terraform-tests
+    url: https://api.scaleway.com/functions/v1beta1/regions/fr-par/namespaces
+    method: POST
+  response:
+    body: '{"id":"aaf055f2-5815-446d-8f51-883fa9a4980f","name":"tf-func--sad-wilson","environment_variables":{},"organization_id":"ee7bd9e1-9cbd-4724-b2f4-19e50f3cf38b","project_id":"ee7bd9e1-9cbd-4724-b2f4-19e50f3cf38b","status":"pending","registry_namespace_id":"","error_message":null,"registry_endpoint":"","description":"","secret_environment_variables":[],"region":"fr-par"}'
+    headers:
+      Content-Length:
+      - "369"
+      Content-Security-Policy:
+      - default-src 'none'; frame-ancestors 'none'
+      Content-Type:
+      - application/json
+      Date:
+      - Tue, 20 Sep 2022 15:36:38 GMT
+      Server:
+      - Scaleway API-Gateway
+      Strict-Transport-Security:
+      - max-age=63072000
+      X-Content-Type-Options:
+      - nosniff
+      X-Frame-Options:
+      - DENY
+      X-Request-Id:
+      - e45c9e70-97d6-4559-bb22-69fc951d9d2d
+    status: 200 OK
+    code: 200
+    duration: ""
+- request:
+    body: ""
+    form: {}
+    headers:
+      User-Agent:
+      - scaleway-sdk-go/v1.0.0-beta.7+dev (go1.19.1; linux; amd64) terraform-provider/develop
+        terraform/terraform-tests
+    url: https://api.scaleway.com/functions/v1beta1/regions/fr-par/namespaces/aaf055f2-5815-446d-8f51-883fa9a4980f
+    method: GET
+  response:
+    body: '{"id":"aaf055f2-5815-446d-8f51-883fa9a4980f","name":"tf-func--sad-wilson","environment_variables":{},"organization_id":"ee7bd9e1-9cbd-4724-b2f4-19e50f3cf38b","project_id":"ee7bd9e1-9cbd-4724-b2f4-19e50f3cf38b","status":"pending","registry_namespace_id":"","error_message":null,"registry_endpoint":"","description":"","secret_environment_variables":[],"region":"fr-par"}'
+    headers:
+      Content-Length:
+      - "369"
+      Content-Security-Policy:
+      - default-src 'none'; frame-ancestors 'none'
+      Content-Type:
+      - application/json
+      Date:
+      - Tue, 20 Sep 2022 15:36:38 GMT
+      Server:
+      - Scaleway API-Gateway
+      Strict-Transport-Security:
+      - max-age=63072000
+      X-Content-Type-Options:
+      - nosniff
+      X-Frame-Options:
+      - DENY
+      X-Request-Id:
+      - 6bc6bf9d-733d-4944-bc1a-8208284b6556
+    status: 200 OK
+    code: 200
+    duration: ""
+- request:
+    body: ""
+    form: {}
+    headers:
+      User-Agent:
+      - scaleway-sdk-go/v1.0.0-beta.7+dev (go1.19.1; linux; amd64) terraform-provider/develop
+        terraform/terraform-tests
+    url: https://api.scaleway.com/functions/v1beta1/regions/fr-par/namespaces/aaf055f2-5815-446d-8f51-883fa9a4980f
+    method: GET
+  response:
+    body: '{"id":"aaf055f2-5815-446d-8f51-883fa9a4980f","name":"tf-func--sad-wilson","environment_variables":{},"organization_id":"ee7bd9e1-9cbd-4724-b2f4-19e50f3cf38b","project_id":"ee7bd9e1-9cbd-4724-b2f4-19e50f3cf38b","status":"ready","registry_namespace_id":"3703a249-669f-4e77-a35f-4693d3430c65","error_message":null,"registry_endpoint":"rg.fr-par.scw.cloud/funcscwtffuncsadwilsonzybdszae","description":"","secret_environment_variables":[],"region":"fr-par"}'
+    headers:
+      Content-Length:
+      - "453"
+      Content-Security-Policy:
+      - default-src 'none'; frame-ancestors 'none'
+      Content-Type:
+      - application/json
+      Date:
+      - Tue, 20 Sep 2022 15:36:44 GMT
+      Server:
+      - Scaleway API-Gateway
+      Strict-Transport-Security:
+      - max-age=63072000
+      X-Content-Type-Options:
+      - nosniff
+      X-Frame-Options:
+      - DENY
+      X-Request-Id:
+      - 4dbdec2a-e085-482e-810d-b0b512fbcde9
+    status: 200 OK
+    code: 200
+    duration: ""
+- request:
+    body: ""
+    form: {}
+    headers:
+      User-Agent:
+      - scaleway-sdk-go/v1.0.0-beta.7+dev (go1.19.1; linux; amd64) terraform-provider/develop
+        terraform/terraform-tests
+    url: https://api.scaleway.com/functions/v1beta1/regions/fr-par/namespaces/aaf055f2-5815-446d-8f51-883fa9a4980f
+    method: GET
+  response:
+    body: '{"id":"aaf055f2-5815-446d-8f51-883fa9a4980f","name":"tf-func--sad-wilson","environment_variables":{},"organization_id":"ee7bd9e1-9cbd-4724-b2f4-19e50f3cf38b","project_id":"ee7bd9e1-9cbd-4724-b2f4-19e50f3cf38b","status":"ready","registry_namespace_id":"3703a249-669f-4e77-a35f-4693d3430c65","error_message":null,"registry_endpoint":"rg.fr-par.scw.cloud/funcscwtffuncsadwilsonzybdszae","description":"","secret_environment_variables":[],"region":"fr-par"}'
+    headers:
+      Content-Length:
+      - "453"
+      Content-Security-Policy:
+      - default-src 'none'; frame-ancestors 'none'
+      Content-Type:
+      - application/json
+      Date:
+      - Tue, 20 Sep 2022 15:36:44 GMT
+      Server:
+      - Scaleway API-Gateway
+      Strict-Transport-Security:
+      - max-age=63072000
+      X-Content-Type-Options:
+      - nosniff
+      X-Frame-Options:
+      - DENY
+      X-Request-Id:
+      - 20e3d499-9f98-446f-9597-9d4ab4c9ea3d
+    status: 200 OK
+    code: 200
+    duration: ""
+- request:
+    body: '{"name":"foobar","namespace_id":"aaf055f2-5815-446d-8f51-883fa9a4980f","environment_variables":{},"min_scale":0,"max_scale":20,"runtime":"node14","memory_limit":128,"timeout":"10.000000000s","handler":"handler.handle","privacy":"private","description":null,"secret_environment_variables":null}'
+    form: {}
+    headers:
+      Content-Type:
+      - application/json
+      User-Agent:
+      - scaleway-sdk-go/v1.0.0-beta.7+dev (go1.19.1; linux; amd64) terraform-provider/develop
+        terraform/terraform-tests
+    url: https://api.scaleway.com/functions/v1beta1/regions/fr-par/functions
+    method: POST
+  response:
+    body: '{"id":"d63d50bc-5367-4a0a-bedc-c5433faf672e","name":"foobar","namespace_id":"aaf055f2-5815-446d-8f51-883fa9a4980f","status":"created","environment_variables":{},"min_scale":0,"max_scale":20,"runtime":"node14","memory_limit":128,"cpu_limit":70,"timeout":"10s","handler":"handler.handle","error_message":null,"privacy":"private","description":"","domain_name":"tffuncsadwilsonzybdszae-foobar.functions.fnc.fr-par.scw.cloud","secret_environment_variables":[],"http_option":"","runtime_message":"","region":"fr-par"}'
+    headers:
+      Content-Length:
+      - "512"
+      Content-Security-Policy:
+      - default-src 'none'; frame-ancestors 'none'
+      Content-Type:
+      - application/json
+      Date:
+      - Tue, 20 Sep 2022 15:36:45 GMT
+      Server:
+      - Scaleway API-Gateway
+      Strict-Transport-Security:
+      - max-age=63072000
+      X-Content-Type-Options:
+      - nosniff
+      X-Frame-Options:
+      - DENY
+      X-Request-Id:
+      - 587100f6-4c8f-4141-aa9f-d2843448e4d0
+    status: 200 OK
+    code: 200
+    duration: ""
+- request:
+    body: ""
+    form: {}
+    headers:
+      User-Agent:
+      - scaleway-sdk-go/v1.0.0-beta.7+dev (go1.19.1; linux; amd64) terraform-provider/develop
+        terraform/terraform-tests
+    url: https://api.scaleway.com/functions/v1beta1/regions/fr-par/functions/d63d50bc-5367-4a0a-bedc-c5433faf672e
+    method: GET
+  response:
+    body: '{"id":"d63d50bc-5367-4a0a-bedc-c5433faf672e","name":"foobar","namespace_id":"aaf055f2-5815-446d-8f51-883fa9a4980f","status":"created","environment_variables":{},"min_scale":0,"max_scale":20,"runtime":"node14","memory_limit":128,"cpu_limit":70,"timeout":"10s","handler":"handler.handle","error_message":null,"privacy":"private","description":"","domain_name":"tffuncsadwilsonzybdszae-foobar.functions.fnc.fr-par.scw.cloud","secret_environment_variables":[],"http_option":"","runtime_message":"","region":"fr-par"}'
+    headers:
+      Content-Length:
+      - "512"
+      Content-Security-Policy:
+      - default-src 'none'; frame-ancestors 'none'
+      Content-Type:
+      - application/json
+      Date:
+      - Tue, 20 Sep 2022 15:36:45 GMT
+      Server:
+      - Scaleway API-Gateway
+      Strict-Transport-Security:
+      - max-age=63072000
+      X-Content-Type-Options:
+      - nosniff
+      X-Frame-Options:
+      - DENY
+      X-Request-Id:
+      - e11143a8-76fb-4dd2-b52c-591e7665e1bf
+    status: 200 OK
+    code: 200
+    duration: ""
+- request:
+    body: ""
+    form: {}
+    headers:
+      User-Agent:
+      - scaleway-sdk-go/v1.0.0-beta.7+dev (go1.19.1; linux; amd64) terraform-provider/develop
+        terraform/terraform-tests
+    url: https://api.scaleway.com/functions/v1beta1/regions/fr-par/functions/d63d50bc-5367-4a0a-bedc-c5433faf672e
+    method: GET
+  response:
+    body: '{"id":"d63d50bc-5367-4a0a-bedc-c5433faf672e","name":"foobar","namespace_id":"aaf055f2-5815-446d-8f51-883fa9a4980f","status":"created","environment_variables":{},"min_scale":0,"max_scale":20,"runtime":"node14","memory_limit":128,"cpu_limit":70,"timeout":"10s","handler":"handler.handle","error_message":null,"privacy":"private","description":"","domain_name":"tffuncsadwilsonzybdszae-foobar.functions.fnc.fr-par.scw.cloud","secret_environment_variables":[],"http_option":"","runtime_message":"","region":"fr-par"}'
+    headers:
+      Content-Length:
+      - "512"
+      Content-Security-Policy:
+      - default-src 'none'; frame-ancestors 'none'
+      Content-Type:
+      - application/json
+      Date:
+      - Tue, 20 Sep 2022 15:36:45 GMT
+      Server:
+      - Scaleway API-Gateway
+      Strict-Transport-Security:
+      - max-age=63072000
+      X-Content-Type-Options:
+      - nosniff
+      X-Frame-Options:
+      - DENY
+      X-Request-Id:
+      - ba9378db-c33b-44d1-8b33-4bfb154a5b8c
+    status: 200 OK
+    code: 200
+    duration: ""
+- request:
+    body: ""
+    form: {}
+    headers:
+      User-Agent:
+      - scaleway-sdk-go/v1.0.0-beta.7+dev (go1.19.1; linux; amd64) terraform-provider/develop
+        terraform/terraform-tests
+    url: https://api.scaleway.com/functions/v1beta1/regions/fr-par/functions/d63d50bc-5367-4a0a-bedc-c5433faf672e
+    method: GET
+  response:
+    body: '{"id":"d63d50bc-5367-4a0a-bedc-c5433faf672e","name":"foobar","namespace_id":"aaf055f2-5815-446d-8f51-883fa9a4980f","status":"created","environment_variables":{},"min_scale":0,"max_scale":20,"runtime":"node14","memory_limit":128,"cpu_limit":70,"timeout":"10s","handler":"handler.handle","error_message":null,"privacy":"private","description":"","domain_name":"tffuncsadwilsonzybdszae-foobar.functions.fnc.fr-par.scw.cloud","secret_environment_variables":[],"http_option":"","runtime_message":"","region":"fr-par"}'
+    headers:
+      Content-Length:
+      - "512"
+      Content-Security-Policy:
+      - default-src 'none'; frame-ancestors 'none'
+      Content-Type:
+      - application/json
+      Date:
+      - Tue, 20 Sep 2022 15:36:45 GMT
+      Server:
+      - Scaleway API-Gateway
+      Strict-Transport-Security:
+      - max-age=63072000
+      X-Content-Type-Options:
+      - nosniff
+      X-Frame-Options:
+      - DENY
+      X-Request-Id:
+      - eb2cf498-a5d7-4874-aa4e-c5772d847284
+    status: 200 OK
+    code: 200
+    duration: ""
+- request:
+    body: ""
+    form: {}
+    headers:
+      User-Agent:
+      - scaleway-sdk-go/v1.0.0-beta.7+dev (go1.19.1; linux; amd64) terraform-provider/develop
+        terraform/terraform-tests
+    url: https://api.scaleway.com/functions/v1beta1/regions/fr-par/namespaces/aaf055f2-5815-446d-8f51-883fa9a4980f
+    method: GET
+  response:
+    body: '{"id":"aaf055f2-5815-446d-8f51-883fa9a4980f","name":"tf-func--sad-wilson","environment_variables":{},"organization_id":"ee7bd9e1-9cbd-4724-b2f4-19e50f3cf38b","project_id":"ee7bd9e1-9cbd-4724-b2f4-19e50f3cf38b","status":"ready","registry_namespace_id":"3703a249-669f-4e77-a35f-4693d3430c65","error_message":null,"registry_endpoint":"rg.fr-par.scw.cloud/funcscwtffuncsadwilsonzybdszae","description":"","secret_environment_variables":[],"region":"fr-par"}'
+    headers:
+      Content-Length:
+      - "453"
+      Content-Security-Policy:
+      - default-src 'none'; frame-ancestors 'none'
+      Content-Type:
+      - application/json
+      Date:
+      - Tue, 20 Sep 2022 15:36:46 GMT
+      Server:
+      - Scaleway API-Gateway
+      Strict-Transport-Security:
+      - max-age=63072000
+      X-Content-Type-Options:
+      - nosniff
+      X-Frame-Options:
+      - DENY
+      X-Request-Id:
+      - eb5ed353-3d37-4273-887c-56b65ae30026
+    status: 200 OK
+    code: 200
+    duration: ""
+- request:
+    body: ""
+    form: {}
+    headers:
+      User-Agent:
+      - scaleway-sdk-go/v1.0.0-beta.7+dev (go1.19.1; linux; amd64) terraform-provider/develop
+        terraform/terraform-tests
+    url: https://api.scaleway.com/functions/v1beta1/regions/fr-par/functions/d63d50bc-5367-4a0a-bedc-c5433faf672e
+    method: GET
+  response:
+    body: '{"id":"d63d50bc-5367-4a0a-bedc-c5433faf672e","name":"foobar","namespace_id":"aaf055f2-5815-446d-8f51-883fa9a4980f","status":"created","environment_variables":{},"min_scale":0,"max_scale":20,"runtime":"node14","memory_limit":128,"cpu_limit":70,"timeout":"10s","handler":"handler.handle","error_message":null,"privacy":"private","description":"","domain_name":"tffuncsadwilsonzybdszae-foobar.functions.fnc.fr-par.scw.cloud","secret_environment_variables":[],"http_option":"","runtime_message":"","region":"fr-par"}'
+    headers:
+      Content-Length:
+      - "512"
+      Content-Security-Policy:
+      - default-src 'none'; frame-ancestors 'none'
+      Content-Type:
+      - application/json
+      Date:
+      - Tue, 20 Sep 2022 15:36:46 GMT
+      Server:
+      - Scaleway API-Gateway
+      Strict-Transport-Security:
+      - max-age=63072000
+      X-Content-Type-Options:
+      - nosniff
+      X-Frame-Options:
+      - DENY
+      X-Request-Id:
+      - 24cc6813-7076-4a87-836a-ed395040b913
+    status: 200 OK
+    code: 200
+    duration: ""
+- request:
+    body: ""
+    form: {}
+    headers:
+      User-Agent:
+      - scaleway-sdk-go/v1.0.0-beta.7+dev (go1.19.1; linux; amd64) terraform-provider/develop
+        terraform/terraform-tests
+    url: https://api.scaleway.com/functions/v1beta1/regions/fr-par/functions/d63d50bc-5367-4a0a-bedc-c5433faf672e
+    method: GET
+  response:
+    body: '{"id":"d63d50bc-5367-4a0a-bedc-c5433faf672e","name":"foobar","namespace_id":"aaf055f2-5815-446d-8f51-883fa9a4980f","status":"created","environment_variables":{},"min_scale":0,"max_scale":20,"runtime":"node14","memory_limit":128,"cpu_limit":70,"timeout":"10s","handler":"handler.handle","error_message":null,"privacy":"private","description":"","domain_name":"tffuncsadwilsonzybdszae-foobar.functions.fnc.fr-par.scw.cloud","secret_environment_variables":[],"http_option":"","runtime_message":"","region":"fr-par"}'
+    headers:
+      Content-Length:
+      - "512"
+      Content-Security-Policy:
+      - default-src 'none'; frame-ancestors 'none'
+      Content-Type:
+      - application/json
+      Date:
+      - Tue, 20 Sep 2022 15:36:47 GMT
+      Server:
+      - Scaleway API-Gateway
+      Strict-Transport-Security:
+      - max-age=63072000
+      X-Content-Type-Options:
+      - nosniff
+      X-Frame-Options:
+      - DENY
+      X-Request-Id:
+      - e35547d2-722c-45fb-a46b-0c0866ee9104
+    status: 200 OK
+    code: 200
+    duration: ""
+- request:
+    body: ""
+    form: {}
+    headers:
+      User-Agent:
+      - scaleway-sdk-go/v1.0.0-beta.7+dev (go1.19.1; linux; amd64) terraform-provider/develop
+        terraform/terraform-tests
+    url: https://api.scaleway.com/functions/v1beta1/regions/fr-par/functions/d63d50bc-5367-4a0a-bedc-c5433faf672e
+    method: DELETE
+  response:
+    body: '{"id":"d63d50bc-5367-4a0a-bedc-c5433faf672e","name":"foobar","namespace_id":"aaf055f2-5815-446d-8f51-883fa9a4980f","status":"deleting","environment_variables":{},"min_scale":0,"max_scale":20,"runtime":"node14","memory_limit":128,"cpu_limit":70,"timeout":"10s","handler":"handler.handle","error_message":null,"privacy":"private","description":"","domain_name":"tffuncsadwilsonzybdszae-foobar.functions.fnc.fr-par.scw.cloud","secret_environment_variables":[],"http_option":"","runtime_message":"","region":"fr-par"}'
+    headers:
+      Content-Length:
+      - "513"
+      Content-Security-Policy:
+      - default-src 'none'; frame-ancestors 'none'
+      Content-Type:
+      - application/json
+      Date:
+      - Tue, 20 Sep 2022 15:36:47 GMT
+      Server:
+      - Scaleway API-Gateway
+      Strict-Transport-Security:
+      - max-age=63072000
+      X-Content-Type-Options:
+      - nosniff
+      X-Frame-Options:
+      - DENY
+      X-Request-Id:
+      - 74c7db0f-c8fe-4644-9d51-b09a8506b8da
+    status: 200 OK
+    code: 200
+    duration: ""
+- request:
+    body: ""
+    form: {}
+    headers:
+      User-Agent:
+      - scaleway-sdk-go/v1.0.0-beta.7+dev (go1.19.1; linux; amd64) terraform-provider/develop
+        terraform/terraform-tests
+    url: https://api.scaleway.com/functions/v1beta1/regions/fr-par/namespaces/aaf055f2-5815-446d-8f51-883fa9a4980f
+    method: GET
+  response:
+    body: '{"id":"aaf055f2-5815-446d-8f51-883fa9a4980f","name":"tf-func--sad-wilson","environment_variables":{},"organization_id":"ee7bd9e1-9cbd-4724-b2f4-19e50f3cf38b","project_id":"ee7bd9e1-9cbd-4724-b2f4-19e50f3cf38b","status":"ready","registry_namespace_id":"3703a249-669f-4e77-a35f-4693d3430c65","error_message":null,"registry_endpoint":"rg.fr-par.scw.cloud/funcscwtffuncsadwilsonzybdszae","description":"","secret_environment_variables":[],"region":"fr-par"}'
+    headers:
+      Content-Length:
+      - "453"
+      Content-Security-Policy:
+      - default-src 'none'; frame-ancestors 'none'
+      Content-Type:
+      - application/json
+      Date:
+      - Tue, 20 Sep 2022 15:36:47 GMT
+      Server:
+      - Scaleway API-Gateway
+      Strict-Transport-Security:
+      - max-age=63072000
+      X-Content-Type-Options:
+      - nosniff
+      X-Frame-Options:
+      - DENY
+      X-Request-Id:
+      - 97bcfa6c-cb06-41a2-b1df-2f09967cab8d
+    status: 200 OK
+    code: 200
+    duration: ""
+- request:
+    body: ""
+    form: {}
+    headers:
+      User-Agent:
+      - scaleway-sdk-go/v1.0.0-beta.7+dev (go1.19.1; linux; amd64) terraform-provider/develop
+        terraform/terraform-tests
+    url: https://api.scaleway.com/functions/v1beta1/regions/fr-par/namespaces/aaf055f2-5815-446d-8f51-883fa9a4980f
+    method: DELETE
+  response:
+    body: '{"id":"aaf055f2-5815-446d-8f51-883fa9a4980f","name":"tf-func--sad-wilson","environment_variables":{},"organization_id":"ee7bd9e1-9cbd-4724-b2f4-19e50f3cf38b","project_id":"ee7bd9e1-9cbd-4724-b2f4-19e50f3cf38b","status":"deleting","registry_namespace_id":"3703a249-669f-4e77-a35f-4693d3430c65","error_message":null,"registry_endpoint":"rg.fr-par.scw.cloud/funcscwtffuncsadwilsonzybdszae","description":"","secret_environment_variables":[],"region":"fr-par"}'
+    headers:
+      Content-Length:
+      - "456"
+      Content-Security-Policy:
+      - default-src 'none'; frame-ancestors 'none'
+      Content-Type:
+      - application/json
+      Date:
+      - Tue, 20 Sep 2022 15:36:48 GMT
+      Server:
+      - Scaleway API-Gateway
+      Strict-Transport-Security:
+      - max-age=63072000
+      X-Content-Type-Options:
+      - nosniff
+      X-Frame-Options:
+      - DENY
+      X-Request-Id:
+      - b1f632f7-0290-4db2-a12f-7213c52c1cf3
+    status: 200 OK
+    code: 200
+    duration: ""
+- request:
+    body: ""
+    form: {}
+    headers:
+      User-Agent:
+      - scaleway-sdk-go/v1.0.0-beta.7+dev (go1.19.1; linux; amd64) terraform-provider/develop
+        terraform/terraform-tests
+    url: https://api.scaleway.com/functions/v1beta1/regions/fr-par/namespaces/aaf055f2-5815-446d-8f51-883fa9a4980f
+    method: GET
+  response:
+    body: '{"message":"Namespace was not found"}'
+    headers:
+      Content-Length:
+      - "37"
+      Content-Security-Policy:
+      - default-src 'none'; frame-ancestors 'none'
+      Content-Type:
+      - application/json
+      Date:
+      - Tue, 20 Sep 2022 15:36:48 GMT
+      Server:
+      - Scaleway API-Gateway
+      Strict-Transport-Security:
+      - max-age=63072000
+      X-Content-Type-Options:
+      - nosniff
+      X-Frame-Options:
+      - DENY
+      X-Request-Id:
+      - 98b9d01f-eb18-4017-8d76-8ff13287b9b3
+    status: 404 Not Found
+    code: 404
+    duration: ""
+- request:
+    body: ""
+    form: {}
+    headers:
+      User-Agent:
+      - scaleway-sdk-go/v1.0.0-beta.7+dev (go1.19.1; linux; amd64) terraform-provider/develop
+        terraform/terraform-tests
+    url: https://api.scaleway.com/functions/v1beta1/regions/fr-par/functions/d63d50bc-5367-4a0a-bedc-c5433faf672e
+    method: DELETE
+  response:
+    body: '{"message":"Function was not found"}'
+    headers:
+      Content-Length:
+      - "36"
+      Content-Security-Policy:
+      - default-src 'none'; frame-ancestors 'none'
+      Content-Type:
+      - application/json
+      Date:
+      - Tue, 20 Sep 2022 15:36:48 GMT
+      Server:
+      - Scaleway API-Gateway
+      Strict-Transport-Security:
+      - max-age=63072000
+      X-Content-Type-Options:
+      - nosniff
+      X-Frame-Options:
+      - DENY
+      X-Request-Id:
+      - c83f9ae1-5d88-49fb-9d0d-a6326267d74b
+    status: 404 Not Found
+    code: 404
+    duration: ""

